inifix/format.py

```diff
@@ -2,44 +2,80 @@
 import os
 import re
 import sys
+import warnings
 from typing import Optional
 from typing import Sequence
 
 from inifix.io import load
 
 NAME_COL_MINSIZE = 10
 COMMENT_COL = 80
+PADDING_SIZE = 4
+
+
+def _ljust(s: str, width: int) -> str:
+    """
+    A more flexible str.ljust
+    Guarantees the return value has at least one trailing whitespace char,
+    even at the cost of going over the specified width.
+    """
+    justified_name = s.ljust(width)
+    if len(s) < width:
+        readable_name = justified_name
+    else:
+        readable_name = justified_name + " "
+    return readable_name
 
 
 def _normalize_whitespace(s: str) -> str:
     return re.sub(r"\s", " ", s).strip()
 
 
-def iniformat(data: str) -> str:
+def iniformat(data: str, *, column_length: Optional[int] = None) -> str:
     lines = data.splitlines()
     contents = []
     comments = []
+    parameters = []
+    values = []
     for line in lines:
         content, _, comment = line.partition("#")
         contents.append(_normalize_whitespace(content))
         comments.append(_normalize_whitespace(comment))
+        if not content.startswith("[") and content != "":
+            parameter, *value = content.split()
+            parameters.append(parameter)
+            values.append(value)
+
+    max_name_size = max(len(parameter) for parameter in parameters)
+
+    if column_length is None:
+        padded_name_col_size = max(max_name_size + PADDING_SIZE, NAME_COL_MINSIZE)
+    else:
+        padded_name_col_size = column_length
+
+        if max_name_size >= column_length:
+            long_parameters = [p for p in parameters if len(p) >= column_length]
+            warnings.warn(
+                "parameter/s:\n"
+                + "\n".join(long_parameters)
+                + "are longer than user specified column length ({column_length}). "
+                + "Column length will be adjusted regardless.",
+            )
 
-    # comment_col = max(80, len(max(contents)) + 10)
-    max_name_size = max(
-        len(c.split()[0]) for c in contents if c and not c.startswith("[")
-    )
-    name_col_size = max(max_name_size + 4, NAME_COL_MINSIZE)
     new_lines = []
+    parameter_count = 0
     for content, comment in zip(contents, comments):
         new_line = ""
         offset = 0
         if content:
             if content.startswith("["):
                 new_line += f"{content}\n"
             else:
-                name, *values = content.split()
-                new_line += f"{name.ljust(name_col_size)}    {'  '.join(values)}"
+                new_line += _ljust(parameters[parameter_count], padded_name_col_size)
+                new_line += "  ".join(values[parameter_count])
+
                 offset = COMMENT_COL - len(new_line)
+                parameter_count += 1  # all the lists are not of the same length : len(parameters) == len(values) != len(contents) == len(comments)
```

Comment:
Actually I found this comment more confusing that helping :/
```suggestion
                parameter_idx+= 1
```